src/main/java/org/broadinstitute/hellbender/tools/spark/sv/StructuralVariantContext.java

```diff
@@ -0,0 +1,280 @@
+        System.arraycopy(referenceBaseBytes, leftPaddingSize + deletionSize, resultBases, leftPaddingSize, rightPaddingSize);
+        final Cigar cigar = new Cigar(Arrays.asList(new CigarElement(leftPaddingSize, CigarOperator.M),
+                new CigarElement(deletionSize, CigarOperator.D),
+                new CigarElement(rightPaddingSize, CigarOperator.M)));
+        final Haplotype result = new Haplotype(resultBases, false);
+        result.setCigar(cigar);
+        result.setGenomeLocation(referenceBases.getInterval());
+        return result;
+    }
+
+    private Haplotype composeInsertionHaplotype(final ReferenceBases referenceBases) {
+        final byte[] insertedSequence = getInsertedSequence();
+        final byte[] referenceBaseBytes = referenceBases.getBases();
+        final byte[] resultBases = new byte[referenceBases.getInterval().size() + insertedSequence.length];
+        final int leftPaddingSize = getStart() - referenceBases.getInterval().getStart() + 1;
+        final int rightPaddingSize = referenceBases.getInterval().getEnd() - getStart();
+        System.arraycopy(referenceBaseBytes, 0, resultBases, 0, leftPaddingSize);
+        System.arraycopy(insertedSequence, 0, resultBases, leftPaddingSize, insertedSequence.length);
+        System.arraycopy(referenceBaseBytes, leftPaddingSize    , resultBases, leftPaddingSize + insertedSequence.length, rightPaddingSize);
+        final Cigar cigar = new Cigar(Arrays.asList(new CigarElement(leftPaddingSize, CigarOperator.M),
+                new CigarElement(insertedSequence.length, CigarOperator.I),
+                new CigarElement(rightPaddingSize, CigarOperator.M)));
+        final Haplotype result = new Haplotype(resultBases, false);
+        result.setCigar(cigar);
+        result.setGenomeLocation(referenceBases.getInterval());
+        return result;
+    }
+
+    /**
+     * Returns the inserted sequence.
+     * <p>
+     *     Currently this method relies on the annotation {@value GATKSVVCFConstants#INSERTED_SEQUENCE}.
+     * </p>
+     * @return {@code null} if there is no inserted sequence.
+     */
+    public byte[] getInsertedSequence() {
+        if (hasAttribute(GATKSVVCFConstants.INSERTED_SEQUENCE)) {
+            final String asString = getAttributeAsString(GATKSVVCFConstants.INSERTED_SEQUENCE, null);
+            if (asString == null) {
+                return null;
+            } else {
+                return asString.getBytes();
+            }
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Returns the absolute structural variant length as recorded in the SVLEN annotation.
+     * <p>
+     *     This method always return a 0 or positive value, so for example for a deletion of 10bp
+     *     it will return 10 despite that the SVLEN annotation should be -10.
+     * </p>
+     * @return -1 if there is no SVLEN annotation, 0 or greater otherwise.
```

Comment:
Not by my understanding. The VCF spec says that the length is the difference between reference allele and alt allele, so technically for a simple inversion, the length would also be 0, although there are different opinions about this definition. So personally I prefer to have no `END` and `SVLEN` at all for BND records (they have `MATE`'s). We could, for the moment, have a static const specifying a value to indicate that `END` and `SVLEN` don't apply for certain records.